client/signup/main.jsx

```diff
@@ -582,9 +586,10 @@ class Signup extends Component {
 				intent,
 				startingPoint,
 				isBlankCanvas: isBlankCanvasDesign( dependencies.selectedDesign ),
-				isMapping: domainItem && isDomainMapping( domainItem ),
-				isTransfer: domainItem && isDomainTransfer( domainItem ),
-				signupDomainOrigin: signupDomainOrigin ?? SIGNUP_DOMAIN_ORIGIN.NOT_SET,
+				isMapping: isMapping,
+				isTransfer: isTransfer,
+				signupDomainOrigin:
+					isTransferOrMapping || signupDomainOrigin || SIGNUP_DOMAIN_ORIGIN.NOT_SET,
```

Comment:
Maybe
```suggestion
				signupDomainOrigin:
					isTransfer || isMapping
						? SIGNUP_DOMAIN_ORIGIN.USE_YOUR_DOMAIN
						: signupDomainOrigin ?? SIGNUP_DOMAIN_ORIGIN.NOT_SET,
```

Or something similar, I didn't test it.